### DISCUSSION TITLE
v-model.trim not working as expected

### DISCUSSION THREAD

**Discussion by pzgz** [arco-design/arco-design-vue#1803] [Q&A] [answered]
- [ ] I'm sure this does not appear in [the issue list of the repository](https://github.com/arco-design/arco-design-vue/issues)

## Basic Info 
- **Package Name And Version:** @arco-design/web-vue@2.37.4
- **Browser:** chrome106.0.0.0
- **Reproduction Link:** https://codesandbox.io/s/headless-flower-0gfqzo?file=/src/App.vue

## What is expected?
使用`v-model.trim`应该在离开文本款时才触发对录入文本的trim操作，而不是`input`的时候进行trim

## Steps to reproduce
采用`v-model.trim`来绑定modelValue，希望去掉input中输入字符串前后的空格，这个场景还是比较常见的，但一旦加上`.trim`后，会导致无法输入空格，可以看repolink里面的内容，对比的是原生的input tag，工作就是正常的，估计是因为触发了`input`事件，尝试加上`.lazy`，也没有效果

**Comment by luo0412** [comment, marked as answer]
感觉这种应该封装一个自定义的指令 v-trim，和组件库没啥关系，网上抄的还蛮好用的...

https://codesandbox.io/s/flamboyant-tamas-n0z7ei

```vue
<a-input v-trim v-model="val" />
```

**Comment by pzgz** [comment]
> 感觉这种应该封装一个自定义的指令 v-trim，和组件库没啥关系，网上抄的还蛮好用的...
> 
> https://codesandbox.io/s/flamboyant-tamas-n0z7ei
> 
> ```
> <a-input v-trim v-model="val" />
> ```

是的，我现在也是在用自己做的指令，但想到`trim`是vue自带的一个修饰符，理应支持，而且如果从`input`和`change`的角度来考虑的话，至少加上`lazy`也应该有效

**Comment by tazyong** [comment]
> > 感觉这种应该封装一个自定义的指令 v-trim，和组件库没啥关系，网上抄的还蛮好用的...
> > [codesandbox.io/s/flamboyant-tamas-n0z7ei](https://codesandbox.io/s/flamboyant-tamas-n0z7ei)
> > ```
> > <a-input v-trim v-model="val" />
> > ```
> 
> 是的，我现在也是在用自己做的指令，但想到`trim`是vue自带的一个修饰符，理应支持，而且如果从`input`和`change`的角度来考虑的话，至少加上`lazy`也应该有效

没有理应支持的说法，Arco 组件对 vue 来说就是自定义组件，自定义组件要使用修饰符都需要额外定义的，目前 Arco 表单组件没有定义修饰符，所以用不了是正常的。你可以自己修改源码，然后 PR 支持
相关文档：https://cn.vuejs.org/guide/components/events.html#usage-with-v-model
